internal/hook/hook_tracker.go

```diff
@@ -48,37 +48,42 @@ type hookTrackerKey struct {
 	container string
 }
 
-// hookTrackerVal records the execution status of a specific hook.
-// hookTrackerVal is extensible to accommodate additional fields as needs develop.
-type hookTrackerVal struct {
+// hookStatus records the execution status of a specific hook.
+// hookStatus is extensible to accommodate additional fields as needs develop.
+type hookStatus struct {
 	// HookFailed indicates if hook failed to execute.
 	hookFailed bool
 	// hookExecuted indicates if hook already execute.
 	hookExecuted bool
+	// hookErr records error if hook execution fails
+	hookErr error
 }
 
 // HookTracker tracks all hooks' execution status
+// HookTracker is a map that uses the backup/restore name as the key
+// and stores another map representing a group of hooks associated with a specific backup/restore.
```

Comment:
To reflect on our discussion regarding skipped hooks, per my understanding, if a hook is skipped it should not be put into the tracker, correct?

Reply:
Currently skipped hooks are also counted as failed hook along with a detailed error message indicating hooks fails due to being skipped. Those hooks are also put into tracker.

Reply:
Thanks